kernel/interrupts/src/aarch64/mod.rs

```diff
@@ -48,7 +48,7 @@ const MAX_IRQ_NUM: usize = 256;
 // it's an array of function pointers which are meant to handle IRQs.
 // Synchronous Exceptions (including syscalls) are not IRQs on aarch64;
 // this crate doesn't expose any way to handle them at the moment.
-static IRQ_HANDLERS: IrqSafeRwLock<[InterruptHandler; MAX_IRQ_NUM]> = IrqSafeRwLock::new([default_irq_handler; MAX_IRQ_NUM]);
+static IRQ_HANDLERS: IrqSafeRwLock<[Option<InterruptHandler>; MAX_IRQ_NUM]> = IrqSafeRwLock::new([None; MAX_IRQ_NUM]);
```

Comment:
i have no idea; it would make sense that they're non-null but i don't know if that optimization actually exists and is used in the compiler. You can check the size of a fn ptr vs the size of Option<fn ptr> to verify that.

Reply:
Looks like the optimization is working: [Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b85d7763574fe18b4960ac773d5ac7dc)

Reply:
Sounds good to me, I prefer this since it more explicitly expresses our intent. Wish we could do that on x86 too.... but we don't control the format of the IDT.